### PR TITLE
new package bz3 (bzip3)

### PR DESCRIPTION
**Pull request by saltedcoffii**
## Description
`bzip3` is described by its programmer as a spiritual successor to `bzip2`. It has quite impressive compression ratios on random data. In this screenshot, `foo` is 10 MB of randomly generated data from `/dev/random` compared against multiple compression algorithms using their highest compression ratios:

image

It's certainly too experimental to use as Chromebrew's compression algorithm right now, but it's certainly a package we should add.

## Additional information

Works properly (tested in containers):
- [x] `x86_64`
- [x] `i686`
- [x] `armv7l`

#### Run the following to get this pull request's changes locally for testing.
```
CREW_TESTING_REPO=https://github.com/saltedcoffii/chromebrew.git CREW_TESTING_BRANCH=bzip3_1.1.6 CREW_TESTING=1 crew update
```

### CODE REVIEW

File: packages/bz3.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    end
+
+    # Add support for most
+    FileUtils.cp 'bz3less', 'bz3most'
+    system "sed -i 's:less:most:g' bz3most"
+    system "sed -i 's: bz3less : bz3less bz3most :g' Makefile.am"
+
+    FileUtils.cp 'bz3less.1', 'bz3most.1'
+    system "sed -i 's:less:most:g' bz3most.1"
+    system "sed -i 's: bz3less.1 : bz3less.1 bz3most.1 :g' Makefile.am"
```
Comment (uberhacker) [inline packages/bz3.rb:31]: Good call!  Would it make sense to add most support instead of removing less support?

File: packages/bz3.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+class Bz3 < Package
+  description 'bzip3 is a better and stronger spiritual successor to bzip2.'
+  homepage 'https://github.com/kspalaiologos/bzip3'
+  @_ver = '1.1.6'
+  version @_ver
+  license 'LGPL-3'
+  compatibility 'all'
+  source_url 'https://github.com/kspalaiologos/bzip3.git'
+  git_hashtag @_ver
+
```
Comment (uberhacker) [inline packages/bz3.rb:12]: Do you think you can add the binaries with @satmandu's scripts?  To upload and add the binaries, see `tools/gl.sh`.

**Comment by uberhacker** [comment]
Rubocop results:
```
packages/bz3.rb:20:5: C: [Correctable] Style/WordArray: Use %w or %W for an array of words.
    ["bz3cat", "bz3grep", "bz3less", "bz3more", "unbzip3"].each do |file|
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
packages/bz3.rb:20:6: C: [Correctable] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    ["bz3cat", "bz3grep", "bz3less", "bz3more", "unbzip3"].each do |file|
     ^^^^^^^^
packages/bz3.rb:20:16: C: [Correctable] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    ["bz3cat", "bz3grep", "bz3less", "bz3more", "unbzip3"].each do |file|
               ^^^^^^^^^
packages/bz3.rb:20:27: C: [Correctable] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    ["bz3cat", "bz3grep", "bz3less", "bz3more", "unbzip3"].each do |file|
                          ^^^^^^^^^
packages/bz3.rb:20:38: C: [Correctable] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    ["bz3cat", "bz3grep", "bz3less", "bz3more", "unbzip3"].each do |file|
                                     ^^^^^^^^^
packages/bz3.rb:20:49: C: [Correctable] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
    ["bz3cat", "bz3grep", "bz3less", "bz3more", "unbzip3"].each do |file|
                                                ^^^^^^^^^
packages/bz3.rb:34:1: C: [Correctable] Layout/EmptyLines: Extra blank line detected.
packages/bz3.rb:[35](https://github.com/chromebrew/chromebrew/actions/runs/3263815589/jobs/5363462551#step:4:36):3: C: [Correctable] Layout/EmptyLineBetweenDefs: Expected 1 empty line between method definitions; found 2.
  def self.build
  ^^^^^^^^^^^^^^
```

File: packages/bz3.rb [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  depends_on 'glibc' # R
+
+  def self.patch
+    # Remove autoreconf from bootstrap.sh: we'll run it ourselves
+    system "sed -i 's:^autoreconf:#autoreconf:g' bootstrap.sh"
+
+    # Replace /bin/sh shebangs with /usr/bin/env sh ones
+    ["bz3cat", "bz3grep", "bz3less", "bz3more", "unbzip3"].each do |file|
+      system "sed -i 's:^#!/bin/sh:#!/usr/bin/env sh:' #{file}"
```
[the remark this answers is not in the document]
Comment (saltedcoffii) [reply packages/bz3.rb:21]: in case the user has a newer version of dash or something installed

File: packages/bz3.rb (hunk 2 above)
Comment (saltedcoffii) [reply packages/bz3.rb:12]: I don't have a key and don't seem to have permission to make one

File: packages/bz3.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    system './bootstrap.sh'
+    system 'autoreconf -fiv'
+    system "./configure #{CREW_OPTIONS} \
+              --disable-arch-native"
+    system 'make'
+  end
+
+  def self.check
+    # At the time of writing (17 Oct 2022), bzip3 has no checks
+    system 'make', 'check'
```
[the remark this answers is not in the document]
Comment (saltedcoffii) [reply packages/bz3.rb:45]: no, it just passes

File: packages/bz3.rb (hunk 2 above)
Comment (uberhacker) [reply packages/bz3.rb:12]: I DM'd a deploy token in Slack.

**Review by satmandu — APPROVED** [review]

**Pull request merged**